libs/libvtrutil/src/vtr_vector.h

```diff
@@ -157,16 +156,31 @@ class vector : private std::vector<V, Allocator> {
     }
 
     /**
-     * @brief Returns a range containing the key-value pairs.
+     * @brief Provides an iterable object to enumerate the vector.
      *
-     * This function returns a range object that represents the sequence of key-value
-     * pairs within the vector. The range can be used to iterate over the pairs using
-     * standard range-based loops or algorithms.
+     * This function allows for easy enumeration, yielding a tuple of (index, element)
+     * pairs in each iteration. It is similar in functionality to Python's `enumerate()`.
      *
-     * @return A `pair_range` object representing the range of key-value pairs.
+     * @ return An iterable wrapper that can be used in a range-based for loop to obtain
+     * (index, element) pairs.
```

Comment:
I think it would be a good idea to give a short example of how to use this iterator (write the outline of a range-based for and extracting keys & values in the loop body).